src/analysis/fourier/add_circle.lean

```diff
@@ -312,6 +312,49 @@ begin
     ←smul_assoc, smul_eq_mul, one_div_mul_cancel hT.out.ne', one_smul],
 end
 
+lemma fourier_coeff.const_smul (f : add_circle T → E) (c : ℂ) (n : ℤ) :
+  fourier_coeff (c • f) n = c • fourier_coeff f n :=
+by simp_rw [fourier_coeff, pi.smul_apply, ←smul_assoc, smul_eq_mul, mul_comm, ←smul_eq_mul,
+  smul_assoc, integral_smul]
+
+lemma fourier_coeff.const_mul (f : add_circle T → ℂ) (c : ℂ) (n : ℤ) :
+  fourier_coeff (λ x, c * f x) n = c * fourier_coeff f n :=
+begin
+  convert fourier_coeff.const_smul f c n,
+end
+
+omit hT
+
+/-- For a function on `ℝ`, the Fourier coefficients of `f` on `[a, b]` are defined as the
+Fourier coefficients of the unique periodic function agreeing with `f` on `Ioc a b`. -/
+def fourier_coeff_on {a b : ℝ} (hab : a < b) (f : ℝ → E) (n : ℤ) : E :=
```

Comment:
This should be mentioned in the file-level docstring. As well as `fourier_coeff`, by the way.

Reply:
It is now (and `fourier_coeff` was already there)